src/module_1/module_1_meteo_api.py

```diff
@@ -1,7 +1,136 @@
+    """
+    coordinates = COORDINATES.get(city)
+
+    if coordinates is None:
+        print(f"Error: Coordenadas no encontradas para la ciudad {city}")
+        return None
+    
+    latitude, longitude = coordinates['latitude'], coordinates['longitude']
+
+    params ={
+        "latitude": latitude,
+        "longitude": longitude,
+        "start_date": start_date,
+        "end_date": end_date,
+        "models": ["CMCC_CM2_VHR4", "FGOALS_f3_H", "HiRAM_SIT_HR", "MRI_AGCM3_2_S", "EC_Earth3P_HR", "MPI_ESM1_2_XR", "NICAM16_8S"],
+        "daily": VARIABLES
+        }
+
+    data = make_api_request(API_URL, params)
+
+    df = process_climate_data(data)
+
+    return data
+
+
+def process_climate_data(data: Dict) -> pd.DataFrame:
+    """
+    Processes climate data and returns a DataFrame.
+    :param data: Climate data in JSON format
+    :return: Processed DataFrame
+    """
+    daily_data = data.get('daily', {})
+
+    # Crea un DataFrame a partir de los datos diarios
+    df = pd.DataFrame(daily_data)
+
+    # Añade una columna para el tiempo
+    df['time'] = pd.to_datetime(df['time'], format='%Y-%m-%d')
+    df['time'] = df['time'].dt.year
+
+    # Agrupa por año y calcula la media para cada columna
+    df_anual_media = df.groupby(df['time']).mean()
+    df_anual_media.columns = [col + '_media_anual' if col != 'time' else col for col in df_anual_media.columns]
+
+    # Agrupa por año y calcula la std para cada columna
+    df_anual_std = df.groupby(df['time']).std()
+    df_anual_std.columns = [col + '_std_anual' if col != 'time' else col for col in df_anual_std.columns]
+
+    # Hago el dataframe final
+    df = pd.concat((df_anual_media, df_anual_std), axis = 1)
+
+    df.reset_index(inplace=True)
+
+    return df
+
+
+def plot_model_data(parameter: str, models: List[str], df: pd.DataFrame) -> None:
+    """
+    Generates a comparative plot of the mean and standard deviation of a parameter for different models.
+    :param parameter: Parameter to plot
+    :param models: List of models to include in the plot
+    :param df: DataFrame with processed climate data
+    """
+    plt.figure(figsize=(10, 6))
+
+    for model in models:
+        mean_col = f"{parameter}_{model}_media_anual"
+        std_col = f"{parameter}_{model}_std_anual"
+
+        plt.plot(df['time'], df[mean_col], label=f"{model} - Mean")
+        plt.fill_between(df['time'], df[mean_col] - df[std_col], df[mean_col] + df[std_col], alpha=0.3, label=f"{model} - Std")
+
+    plt.title(f'Mean and Standard Deviation of {parameter} Over Time for Different Models')
+    plt.xlabel('Time')
+    plt.ylabel(parameter)
+    plt.legend()
+    plt.show()
+
 
 def main():
-    raise NotImplementedError
+    parameter_to_plot = "temperature_2m_mean"
+    models_to_plot = ["CMCC_CM2_VHR4", "FGOALS_f3_H", "HiRAM_SIT_HR", "MRI_AGCM3_2_S", "EC_Earth3P_HR", "MPI_ESM1_2_XR", "NICAM16_8S"]
+    city = "Madrid"
+    
+    # Function to extract data from API owing to the city selected
```

Comment:
As before, this comments aren't needed and therefore should be removed. Let's focus on picking good, descriptive names for our functions so they are self-explanatory instead.